tests/benchmark/test_anta.py

```diff
@@ -0,0 +1,106 @@
+from pytest_codspeed import BenchmarkFixture
+
+from anta.catalog import AntaCatalog
+from anta.inventory import AntaInventory
+from anta.result_manager import ResultManager
+from anta.result_manager.models import AntaTestStatus
+from anta.runner import main
+
+from .utils import collect, collect_commands
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "inventory",
+    [
+        pytest.param({"count": 1, "disable_cache": True, "reachable": False}, id="1 device"),
+        pytest.param({"count": 2, "disable_cache": True, "reachable": False}, id="2 devices"),
+    ],
+    indirect=True,
+)
+def test_anta_dry_run(benchmark: BenchmarkFixture, catalog: AntaCatalog, inventory: AntaInventory) -> None:
+    """Test and benchmark ANTA in Dry-Run Mode."""
+    # Disable logging during ANTA execution to avoid having these function time in benchmarks
+    logging.disable()
+
+    def bench() -> ResultManager:
+        """Need to wrap the ANTA Runner to instantiate a new ResultManger for each benchmark run."""
+        manager = ResultManager()
+        asyncio.run(main(manager, inventory, catalog, dry_run=True))
+        return manager
+
+    manager = benchmark(lambda: bench())  # pylint: disable=unnecessary-lambda
+    logging.disable(logging.NOTSET)
+    if len(manager.results) != 0:
+        pytest.fail("ANTA Dry-Run mode should not return any result", pytrace=False)
+    if catalog.final_tests_count != len(inventory) * len(catalog.tests):
+        pytest.fail(f"Expected {len(inventory) * len(catalog.tests)} selected tests but got {catalog.final_tests_count}", pytrace=False)
+    bench_info = (
+        "\n--- ANTA NRFU Dry-Run Benchmark Information ---\n" f"Selected tests: {catalog.final_tests_count}\n" "-----------------------------------------------"
+    )
+    logger.info(bench_info)
+
+
+@pytest.mark.parametrize(
+    "inventory",
+    [
+        pytest.param({"count": 1, "disable_cache": True}, id="1 device"),
+        pytest.param({"count": 2, "disable_cache": True}, id="2 devices"),
+    ],
+    indirect=True,
+)
+@patch("anta.models.AntaTest.collect", collect)
+@patch("anta.device.AntaDevice.collect_commands", collect_commands)
+@respx.mock  # Mock eAPI responses
+def test_anta(benchmark: BenchmarkFixture, catalog: AntaCatalog, inventory: AntaInventory) -> None:
+    """Test and benchmark ANTA. Mock eAPI responses."""
+    # Disable logging during ANTA execution to avoid having these function time in benchmarks
+    logging.disable()
+
+    def bench() -> ResultManager:
+        """Need to wrap the ANTA Runner to instantiate a new ResultManger for each benchmark run."""
+        manager = ResultManager()
+        asyncio.run(main(manager, inventory, catalog))
+        return manager
+
+    manager = benchmark(lambda: bench())  # pylint: disable=unnecessary-lambda
+    logging.disable(logging.NOTSET)
+
+    if len(catalog.tests) * len(inventory) != len(manager.results):
+        # This could mean duplicates exist.
```

Comment:
How can we have duplicates? Would adding a unique ID to ResultOverwrite solve this?

Reply:
The code generates an ANTA catalog from the unit test data at tests/units/anta_tests.
The "keys" (more conventional than a real data structure key) is (Type[AntaTest], [unit test name]: str).

When writing this PR, I've caught some copy/paste in those data structures that where duplicated unit tests.
Since we manipulate set() in ANTA code, those duplicated were not ported into the AntaCatalog instance, so I've caught AssertionError that I am letting there for now.

As written in the PR descriptionm, we should consider refactoring these data to use a dictionary with a tuple (Type[AntaTest], : str) as key to avoid duplicates.

Would adding a unique ID to ResultOverwrite solve this? No